modules/aerodyn/src/UnsteadyAero_Driver.f90

```diff
@@ -331,21 +335,29 @@ subroutine setUAinputs(n,u,t,dt,dvrInitInp,timeArr,AOAarr,Uarr,OmegaArr)
 
 
       else
-         indx = min(n,size(timeArr))
-         indx = max(1, indx) ! use constant data at initialization
+         ! check optional variables and allocation status
+         if (any( (/ allocated(timeArr),allocated(AOAarr),allocated(OmegaArr),allocated(Uarr) /) )) then
```

Comment:
Based on the error message below, I think you want to use `all()` instead of `any()` here.